pkg/csi_driver/reconciler.go

```diff
@@ -165,14 +174,349 @@ func (recon *multishareReconciler) reconcileWorker() {
 		}
 	}
 
+	reconstructionStamp := time.Now()
+	klog.V(6).Infof("reconstruction finished in %v", time.Since(instanceListStamp))
+
 	// Assign un-assigned shares to instances; update shareInfo and instanceInfo accordingly,
 	// if there's inconsistency between share and instnace then share has source of truth.
 	recon.assignSharesToInstances(shareInfoMap, instanceInfoMap)
 
+	assignmentStamp := time.Now()
+	klog.V(6).Infof("assignment finished in %v", time.Since(reconstructionStamp))
+
+	ops, err := recon.listMultishareResourceOps(context.TODO())
+	if err != nil {
+		klog.Errorf("error listing ops: %s", err.Error())
+		return
+	}
+
+	opStamp := time.Now()
+	klog.V(6).Infof("List Op finished in %v", time.Since(assignmentStamp))
+
+	recon.sendInstanceRequests(instanceInfoMap, ops)
+
+	instanceReqStamp := time.Now()
+	klog.V(6).Infof("instanceRequest finished in %v", time.Since(opStamp))
+
+	recon.sendShareRequests(instanceInfoMap, shareInfoMap, instanceShares, ops)
+
+	klog.V(6).Infof("shareRequest finished in %v", time.Since(instanceReqStamp))
+
 	klog.Infof("Reconciliation round finished after %v", time.Since(startTime))
 }
 
-func (recon *multishareReconciler) assignSharesToInstances(shareInfos map[string]*v1alpha1.ShareInfo, instanceInfos map[string]*v1alpha1.InstanceInfo) {
+func (recon *MultishareReconciler) sendShareRequests(instanceInfos map[string]*v1alpha1.InstanceInfo, shareInfos map[string]*v1alpha1.ShareInfo, instanceShares map[string][]*file.Share, ops []*Op) {
+	for _, shareInfo := range shareInfos {
+		needDelete := shareInfo.DeletionTimestamp != nil
+
+		if needDelete {
+			if shareInfo.Status == nil {
+				klog.Errorf("shareInfo %s marked for delete but Status subresource is nil", shareInfo.Name)
+				continue
+			}
+			if shareInfo.Status.ShareStatus == v1alpha1.DELETED {
+				klog.Infof("shareInfo status DELETED, no need to send any share request for %s", shareInfo.Name)
```

Comment:
nit: "shareInfo status DELETED, skip sending share request for %s"

Reply:
done